packages/runtime/src/components/page/Page.tsx

```diff
@@ -1,268 +1,35 @@
-  builderEnabled: boolean
-  cleanup: string | null
-  code: string
-  id: string
-  liveEnabled: boolean
-  location: SnippetLocation
-}
-
-const defaultFavicon = {
-  mimetype: 'image/png',
-  publicUrl:
-    'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACIAAAAiCAYAAAA6RwvCAAAACXBIWXMAABcRAAAXEQHKJvM/AAABjElEQVRYhc2XzU3EMBCFB8TddAAXn6EE6GCpgNABZ1/IXnymBOgAOmA7YM8+ABVsXEHQQFaKQryeN3Yk3ilKJtEnv/nLUd/3pFG0riGi88yrnQn+UfJ5FUi0riWiB2H4nQn+KRd0DFP8agXEfkqCYJBoHdtxIQxfm+DfFgEhoith3NYE30o/qgGR2BJB+xY7kdYEL8oNFUi0jiFMJuxVWrJqEMFxsyUNCsE6AeNztvBp7aJ143vXksoRnwhYtmNdSoIQa6RlO9YXEWW7KgoCleOgxgTf1QZBT+RZ2lXFING6UxCCq+ceeUE8fYdknY599v9sJvzGBP+yCEgC7GPmETc0OJ+0awAlkhe2pAbIXAeFZ8xe2g2Nk3c3ub0xwWt6zY9qbmiqGVMbZK21ZC/YmhlbeBMTzZNDQqcvDb1kM1x32iqZSt1HaqukfKvq34BAOTLsrH+ETNmUkKHHA+428RgeclPVWozeSyAI2EdWB34jtqXNTAySOY3i/KgFIlqOa4GkFmBegorzg4joG07he/M7zl6jAAAAAElFTkSuQmCC',
-}
-
-// Taken from https://github.com/facebook/react/blob/14bac6193a334eda42e727336e8967419f08f5df/packages/react-dom/src/server/ReactPartialRenderer.js#L208
-const VALID_TAG_REGEX = /^[a-zA-Z][a-zA-Z:_.\-\d]*$/
-
-const VALID_HEAD_ELEMENT_TYPES = [
-  'title',
-  'base',
-  'link',
-  'style',
-  'meta',
-  'script',
-  'noscript',
-  'template',
-]
-
-function snippetToElement(snippet: Pick<Snippet, 'id' | 'code'>): (string | ReactElement)[] {
-  return Children.map(parse(snippet.code), element => {
-    if (typeof element === 'string') return element
-
-    if (!VALID_TAG_REGEX.test(element.type)) return null
-
-    const key = element.key ? `${snippet.id}:${element.key}` : snippet.id
-
-    return createElement(element.type, { ...element.props, key })
-  })
-}
-
-const filterUsedSnippetProperties = ({
-  code,
-  builderEnabled,
-  liveEnabled,
-  location,
-  cleanup,
-}: Snippet) => ({
-  code,
-  builderEnabled,
-  liveEnabled,
-  location,
-  cleanup,
-})
+import { usePageSnippets } from '../hooks/usePageSnippets'
+import { PageHead } from './PageHead'
 
 type Props = {
   document: MakeswiftPageDocument
 }
 
 export function Page({ document: page }: Props): JSX.Element {
-  const isInBuilder = useIsInBuilder()
-  const [snippets, setSnippets] = useState(page.snippets)
-  // We're using cached results here for page snippets and site fonts so that anytime the user
-  // changes the snippets or fonts on the builder, the change would be reflected here.
-  // See this PR for discussions and things we can do to improve it in the future:
-  // https://github.com/makeswift/makeswift/pull/77
-  const cachedPage = useCachedPage(isInBuilder ? page.id : null)
-  useEffect(() => {
-    if (cachedPage == null) return
-
-    const oldSnippets = snippets.map(filterUsedSnippetProperties)
-    const newSnippets = cachedPage.snippets.map(filterUsedSnippetProperties)
-
-    if (deepEqual(newSnippets, oldSnippets)) return
-
-    setSnippets(cachedPage.snippets)
-  }, [cachedPage])
-  const site = useCachedSite(isInBuilder ? page.site.id : null)
+  const { bodySnippets } = usePageSnippets({ page })
```

Comment:
We seem to be using `usePageSnippets` twice. Once in this component and another time in `PageHead`. Are we worried that this will result in two subscriptions and unnecessary re-renders or potential performance problems?